railties/lib/rails/generators.rb

```diff
@@ -116,6 +116,10 @@ def self.api_only!
         helper: false,
         template_engine: nil
       )
+
+      if ARGV.first == 'mailer'
+        options[:rails].merge!(template_engine: :erb)
```

Comment:
Yes, it can be overridden.. This value is the default value, it is used if it is set to the value in the user configuration.